controller/permissions/permissions.go

```diff
@@ -0,0 +1,178 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package permissions
+
+const (
+	AdminPermission                 = "admin"
+	AdminReadOnlyPermission         = "admin_readonly"
+	AuthenticatedPermission         = "authenticated"
+	PartiallyAuthenticatePermission = "partial_auth"
+)
+
+const Ops = "ops"
+
+type Action string
+
+const (
+	Read   Action = "read"
+	Create Action = "create"
+	Update Action = "update"
+	Delete Action = "delete"
+)
+
+type Api string
+
+const (
+	Client     Api = "client"
+	Management Api = "management"
```

Comment:
Is there a reason we didn't use the API binding values? These should be available from the xweb context stored on the HTTP request context calls.

```
		apiHandler := *xweb.HandlerFromRequestContext(params.HTTPRequest.Context())
		if apiHandler.Binding() == webapis.ManagementApiBinding {
			//...
		}
```

Reply:
It is fine for now. 

This could also have been addressed in the management enforcer code to simply check w/ an `or`/`||` and ensuring the fabric management API has the same context.Context bag throug xweb, which is should.